Remotion/Development/UnitTesting.IsolatedCodeRunner/IsolatedCodeRunner.cs

```diff
@@ -0,0 +1,167 @@
+        : this(testAction, TimeSpan.FromSeconds(15))
+    {
+    }
+
+    public IsolatedCodeRunner (Action<string[]> testAction, TimeSpan timeout)
+    {
+      ArgumentUtility.CheckNotNull(nameof(testAction), testAction);
+
+      _timeout = timeout;
+
+      var testActionMethod = testAction.Method;
+      if (!testActionMethod.IsStatic)
+        throw new ArgumentException("The specified test method must be static.");
+
+      _targetMethod = testActionMethod;
+      _assemblyLocation = testActionMethod.DeclaringType!.Assembly.Location;
+      _typeName = testActionMethod.DeclaringType!.FullName!;
+      _methodName = testActionMethod.Name;
+    }
+
+    public string Output
+    {
+      get
+      {
+        lock (_output)
+          return _output.ToString();
+      }
+    }
+
+    public void Run (params string[] args)
+    {
+      ArgumentUtility.CheckNotNull(nameof(args), args);
+
+      var appProcessRunnerExe = Path.ChangeExtension(typeof(IsolatedCodeRunner).Assembly.Location, "exe");
+      if (!File.Exists(appProcessRunnerExe))
+        throw new InvalidOperationException("Cannot find the isolated code runner exe.");
+
+      var startInfo = new ProcessStartInfo
+                      {
+                          WindowStyle = ProcessWindowStyle.Hidden,
+                          CreateNoWindow = false,
+                          RedirectStandardOutput = true
+                      };
+
+      startInfo.FileName = appProcessRunnerExe;
+      startInfo.Arguments = string.Join(" ", args.Select(e => $@"""{e.Replace("\"", "\"\"")}"""));
+      startInfo.WorkingDirectory = Path.GetDirectoryName(_targetMethod.DeclaringType!.Assembly.Location);
+
+      startInfo.Environment[TargetAssemblyPath] = _assemblyLocation;
+      startInfo.Environment[TargetTypeName] = _typeName;
+      startInfo.Environment[TargetMethodName] = _methodName;
+
+      startInfo.Environment[TargetBaseDirectory] = AppContext.BaseDirectory;
+
+      if (!string.IsNullOrEmpty(ConfigFile))
+        startInfo.Environment[TargetConfigFile] = ConfigFile;
+
+      using var process = new Process();
+      process.StartInfo = startInfo;
+      process.OutputDataReceived += DataReceived;
+
+      process.Start();
+      process.BeginOutputReadLine();
+
+      if (process.WaitForExit((int)_timeout.TotalMilliseconds))
+      {
+        // Need to call .WaitForExit() again to make sure that standard out is flushed
+        process.WaitForExit();
+      }
+      else
+      {
+        process.Kill();
+        throw new InvalidOperationException("Isolated code runner execution failed to stop within the specified timeout.");
+      }
+
+      if (_exception != null)
+        throw _exception;
+
+      if (process.ExitCode != 0)
+        throw new InvalidOperationException($"Isolated code runner execution failed with exit code {process.ExitCode}.");
+    }
+
+    private void DataReceived (object sender, DataReceivedEventArgs args)
```

Comment:
tehre is a lot of stuff in there. if we extract this into a utility class (ExceptionSerializer), we can easiliy test it. or you test it integrated via isolated. it should have 100% coverage, tough.